base/tuple.jl

```diff
@@ -123,6 +123,23 @@ end
 _ntuple(f, n) = (@_noinline_meta; ([f(i) for i = 1:n]...))
 
 # inferrable ntuple
+ntuple{F}(f::F, ::Type{Val{0}}) = (@_inline_meta; ())
```

Comment:
@Sacha0, interesting.  So, did https://github.com/JuliaLang/julia/issues/20282#issuecomment-275801007 happen because the function argument was not called inside the method body?